packages/iTwinUI-react/src/core/TimePicker/TimePicker.test.tsx

```diff
@@ -460,3 +460,444 @@ it('should show values using custom renderers', () => {
   expect(getByText('Before', { selector: '.iui-period li' })).toBeTruthy();
   expect(getByText('After', { selector: '.iui-period li' })).toBeTruthy();
 });
+
+it('should display passed time in combined renderer', () => {
+  const { container } = render(
+    <TimePicker date={new Date(2020, 0, 5, 11, 55)} useCombinedRenderer />,
+  );
+  expect(container.querySelector('.iui-time-picker')).toBeTruthy();
+  const time = container.querySelector(
+    '.iui-time .iui-selected',
+  ) as HTMLElement;
+  expect(time).toBeTruthy();
+  expect(time.textContent).toBe('11:55');
+});
+
+it('should return selected time in combined renderer (seconds)', async () => {
+  const onClick = jest.fn();
+  const { container, getByText } = render(
+    <TimePicker
+      date={new Date(2020, 5, 5, 14, 21, 33)}
+      useCombinedRenderer
+      precision='seconds'
+      onChange={onClick}
+    />,
+  );
+  let selectedTime = container.querySelector(
+    '.iui-time .iui-selected',
+  ) as HTMLElement;
+  expect(selectedTime.textContent).toBe('14:21:33');
+  const newTime = getByText('22:09:22', {
+    selector: '.iui-time li',
+  });
+  await userEvent.click(newTime);
+  expect(onClick).toHaveBeenCalledWith(new Date(2020, 5, 5, 22, 9, 22));
+  selectedTime = container.querySelector(
+    '.iui-time .iui-selected',
+  ) as HTMLElement;
+  expect(selectedTime.textContent).toBe('22:09:22');
+}, 50000);
```

Comment:
![image](https://user-images.githubusercontent.com/37936169/194357807-d9568c00-f955-446f-98ca-f6ff8bfd895f.png)

Reply:
Ohh, sorry, forgot about what you wrote there. I tried looking at lines 32-54 and 333-391 of TimePicker.test.tsx. To confirm, was TimePicker.test.tsx the file you're mentioning?

Reply:
It's okay. But no, Timepicker.tsx

Reply:
> ![image](https://user-images.githubusercontent.com/37936169/194357807-d9568c00-f955-446f-98ca-f6ff8bfd895f.png)

I was reviewing this yesterday. The first point from above looks good as it is but I was thinking about the second point.

I am still thinking about it, but just wondering, would the below code work with slight changes (code is incomplete; only for demo) instead of [L333-389](https://github.com/iTwin/iTwinUI-react/pull/844/files/a39a5b774741fdd12a048918940c5aa9da083945#diff-47435636e77eb38c7aebe52aca573c1aed481fb9e21916cc70c423b6745b88fbR333-R389)? If so, this is lesser lines of code.

The part of I was still thinking about was is there a way to avoid the O(n^3) looking complexity with nested `forEach` loops. But no other way comes to mind at the moment.

```tsx
const data: Date[] = [];

    const possibleSeconds = Array.from(Array(60).keys()).filter(
      (i) => i % secondStep === 0,
    );
    const possibleMinutes = Array.from(Array(60).keys()).filter(
      (i) => i % minuteStep === 0,
    );
    const possibleHours = Array.from(Array(use12Hours ? 12 : 24).keys())
      .filter((i) => i % hourStep === 0)
      .map((i) => (i + 12) % 13);

    possibleHours.forEach((hour) => {
      possibleMinutes.forEach((minute) => {
        possibleSeconds.forEach((second) => {
          data.push(
            new Date(
              time.getFullYear(),
              time.getMonth(),
              time.getDate(),
              hour,
              minute,
              second,
            ),
          );
        });
      });
    });
```